autopush/db.py

```diff
@@ -52,6 +52,15 @@ def hasher(uaid):
     return uaid
 
 
+def normalize_id(id):
+    if not id:
+        return id
+    if (len(id) == 36 and id[8] == id[13] == id[18] == id[23] == '-'):
+        return id.lower()
+    raw = id.lower().replace('-', '')
```

Comment:
This is more a sanity thing. If there are misplaced dashes, strip them before we do any additional formatting. In reality, I should be stripping non-hex characters and checking for string length, but that gets me back precariously to just running the string through uuid.UUID().